cfme/tests/cli/test_appliance_console.py

```diff
@@ -202,3 +207,92 @@ def test_black_console_external_auth(auth_type, app_creds, ipa_crud):
         command_set = ('ap', '', '15', auth_type.index, '4')
     ipa_crud.appliance_console.run_commands(command_set)
     evm_tail.validate_logs()
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
+def test_black_console_external_auth_all(app_creds, ipa_crud):
+    """'ap' launches appliance_console, '' clears info screen, '12/15' change ext auth options,
+    'auth_type' auth type to change, '4' apply changes."""
+
+    evm_tail = LogValidator('/var/www/miq/vmdb/log/evm.log',
+                            matched_patterns=['.*sso_enabled to true.*', '.*saml_enabled to true.*',
+                                '.*local_login_disabled to true.*'],
+                            hostname=ipa_crud.address,
+                            username=app_creds['sshlogin'],
+                            password=app_creds['password'])
+    evm_tail.fix_before_start()
+    if ipa_crud.version >= "5.8":
+        command_set = ('ap', '', '12', '1', '2', '3', '4')
+    else:
+        command_set = ('ap', '', '15', '1', '2', '3', '4')
+    ipa_crud.appliance_console.run_commands(command_set)
+    evm_tail.validate_logs()
+
+    evm_tail = LogValidator('/var/www/miq/vmdb/log/evm.log',
+                            matched_patterns=['.*sso_enabled to false.*',
+                                '.*saml_enabled to false.*', '.*local_login_disabled to false.*'],
+                            hostname=ipa_crud.address,
+                            username=app_creds['sshlogin'],
+                            password=app_creds['password'])
+
+    evm_tail.fix_before_start()
+    if ipa_crud.version >= "5.8":
+        command_set = ('ap', '', '12', '1', '2', '3', '4')
+    else:
+        command_set = ('ap', '', '15', '1', '2', '3', '4')
+    ipa_crud.appliance_console.run_commands(command_set)
+    evm_tail.validate_logs()
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.7')
+def test_black_console_scap(temp_appliance_preconfig, soft_assert):
+    """'ap' launches appliance_console, '' clears info screen, '14/17' Hardens appliance using SCAP
+    configuration, '' complete."""
+
+    if temp_appliance_preconfig.version >= "5.8":
+        command_set = ('ap', '', '14', '')
+    else:
+        command_set = ('ap', '', '17', '')
+    temp_appliance_preconfig.appliance_console.run_commands(command_set)
+
+    with tempfile.NamedTemporaryFile('w') as f:
+        f.write(hidden['scap.rb'])
+        f.flush()
+        os.fsync(f.fileno())
+        temp_appliance_preconfig.ssh_client.put_file(
+            f.name, '/tmp/scap.rb')
+    if temp_appliance_preconfig.version >= "5.8":
+        temp_appliance_preconfig.ssh_client.run_command('cd /tmp/ && ruby scap.rb '
```

Comment:
REQUIRED: I guess you can move some commands out of if statement since those are the same for both 5.7 and 5.8. Also you can use if only for defining different paths not for running commands.

btw: consider defining root_path and other often used paths in appliance and just use them here.